### PR TITLE
Executed excercise 0.4

### PR DESCRIPTION
**Pull request by ArpitAgrawal001**
Here is a simple flow chart:

```mermaid
sequenceDiagram
    participant browser
    participant server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/notes
    activate server
    server-->>browser: HTML document
    deactivate server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/main.css
    activate server
    server-->>browser: the css file
    deactivate server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/main.js
    activate server
    server-->>browser: the JavaScript file
    deactivate server

    Note right of browser: The browser starts executing the JavaScript code that fetches the JSON from the server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/data.json
    activate server
    server-->>browser: [{ "content": "HTML is easy", "date": "2023-1-1" }, ... ]
    deactivate server

    Note right of browser: The browser executes the callback function that renders the notes
    
    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/notes
    activate server
    server-->>browser: Performed the excercise: New note diagram
    deactivate server
    
    Note right of browser: Excercise 0.4
```

### CODE REVIEW

File: Excercise 0.4.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Note right of browser: The browser starts executing the JavaScript code that fetches the JSON from the server
+
+    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/data.json
+    activate server
+    server-->>browser: [{ "content": "HTML is easy", "date": "2023-1-1" }, ... ]
+    deactivate server
+
+    Note right of browser: The browser executes the callback function that renders the notes
+    
+    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/notes
```
Comment (mavenik) [inline Excercise 0.4.md]: @ArpitAgrawal001 Are you sure this is a GET request? You might want to check this.

Also, the sequence diagram does not talk about the response received from the server when a new note is created and actions that follow.

**Review by mavenik — CHANGES_REQUESTED** [review]
The exercise is about understanding a series of request-response cycles that happen when a new note is created. This sequence diagram only shows the request-response cycles when a notes list page is displayed but does not include the network activity when a new note is created. Could you please update your implementation to include those actions as well?

**Comment by ArpitAgrawal001** [comment]
Here is a simple flow chart:

```mermaid
sequenceDiagram
    participant browser
    participant server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/notes
    activate server
    server-->>browser: HTML document
    deactivate server

    browser->>server: sended the user input to the server
    activate server
    server-->>browser: causes the submit event
    deactivate server
    
    browser->>server:Called an Http post request from the URL:https://studies.cs.helsinki.fi/exampleapp/new_note
    activate server
    server-->>browser: responded with HTTP status code 302
    deactivate server

    Note right of browser: The browser reloads the notes page and causes three more HTTP requests.
    

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/main.css
    activate server
    server-->>browser: the css file
    deactivate server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/main.js
    activate server
    server-->>browser: the JavaScript file
    deactivate server

    Note right of browser: The browser starts executing the JavaScript code that fetches the JSON from the server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/data.json
    activate server
    server-->>browser: the raw data of the notes
    deactivate server

    Note right of browser: The browser executes the callback function that renders the notes
    
    browser->>server:Creation of a new element
    activate server
    server-->>browser:Executed the excercise:New note diagram
    deactivate server

    Note right of browser:Excercise 0.4
```

**Comment by ArpitAgrawal001** [comment]
Sir should it be like this?

**Review by mavenik — APPROVED** [review]
Can get better but this does the job.
Approving it 👍

**Comment by ArpitAgrawal001** [comment]
> Can get better but this does the job.
> Approving it 👍 

Sir so should i merge this pr and create a new feature branch for Excercise 0.5

**Comment by mavenik** [comment]
> > Can get better but this does the job.
> > Approving it 👍
> 
> Sir so should i merge this pr and create a new feature branch for Excercise 0.5

Yes @ArpitAgrawal001 

Please merge this into `develop`, then pull latest changes into your local `develop` branch and create a new feature branch from it

````
git checkout develop
git pull origin develop
git checkout -b <new branch name>
````

**Comment by ArpitAgrawal001** [comment]
> > > Can get better but this does the job.
> > > Approving it 👍
> > 
> > Sir so should i merge this pr and create a new feature branch for Excercise 0.5
> 
> Yes @ArpitAgrawal001 
> 
> Please merge this into `develop`, then pull latest changes into your local `develop` branch and create a new feature branch from it
> 
> ````
> git checkout develop
> git pull origin develop
> git checkout -b <new branch name>
> ````

Ok sir

**Pull request merged**